### ISSUE TITLE
Unable to start service com.apple.debugserver.

### ISSUE THREAD

**Issue by firescript** [NativeScript/nativescript-cli#1387]
This only shows up when i `tns run ios --release` or `tns run ios` on my connected device (iPhone 6s). It does not show up when i run on the emulator `tns run ios --emulator`. One thing to note is that the build succeeds on both, emulator and device. It deploys the app to the device properly too. However the CLI seems to crash just after deploying when running on device. The app runs properly after deployment as far as I can tell.

Here is the `tns run ios --release --log trace > run.log`

When the process completes this error message is outputted:

```Error: Unable to start service com.apple.debugserver. Result code is: 3892314146
    at Object.<anonymous> (/usr/local/lib/node_modules/nativescript/lib/common/errors.js:7:23)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/usr/local/lib/node_modules/nativescript/lib/nativescript-cli.js:9:16)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)```

Here is the log.

[run.txt](https://github.com/NativeScript/nativescript-cli/files/85920/run.txt)

This is when i try and run through xcode:

<img width="1151" alt="screen shot 2016-01-11 at 10 59 58 am" src="https://cloud.githubusercontent.com/assets/1789978/12238557/8d3f609a-b852-11e5-86dd-fd8aaebe1417.png">

**Comment by Mitko-Kerezov** [comment]
Hi, @firescript the error code you've received `3892314146` is an invalid service error. 
Could you try and restart your device and see if you get that error again - this usually fixes such errors.

**Comment by firescript** [comment]
Thanks @Mitko-Kerezov, that did the trick!

**Comment by firescript** [comment]
@Mitko-Kerezov it seems to still shows the error after rebooting the phone.

<img width="806" alt="screen shot 2016-01-12 at 11 55 21 am" src="https://cloud.githubusercontent.com/assets/1789978/12270308/675b1e9c-b923-11e5-8f98-5bd7b1804f85.png">

**Comment by rosen-vladimirov** [comment]
Hi, can you try 
```
tns run ios --release
```
and after the crash occurs, please execute
```
tns device log > deviceLog.txt
```
and send us the device log. Also what happens when you try to run the app on the device on your own - as far as I see from the logs, it should be successfully deployed. Are you able to start it manually?

**Comment by firescript** [comment]
Yes I am able to start it manually. It creates the app on the phone. Just no console.log() is able to be done, since the its lost its connection to the app.

[deviceLog.txt](https://github.com/NativeScript/nativescript-cli/files/87676/deviceLog.txt)

**Comment by Mitko-Kerezov** [comment]
Hi @firescript 
We are not able to reproduce the issue on our side but here's what you could try:

1. Can you try running the app in a different device and see if you still get the error?
2. You can also try starting the app through xcode and see if it runs successfully from there (please, note that if you are using Cocoa Pods you'd have to open the workspace (.xcworkspace file) instead of a single project
3. If those fail to give you any insight about the issue, then you could try and send me a sample application reproducing the issue so we can research it on our side. You could use my personal e-mail for this purpose.

Hope this helps

**Comment by firescript** [comment]
When I try and run through Xcode I got "Unable to find developer disk image". So I found this answer and updated to Xcode 7.2 and now it seems to be running

http://stackoverflow.com/a/34649018

Now when I `tns run ios --release` it works and deploys to the phone, however if `tns build ios` or `tns debug ios --emulator` the build fails and I get this error message:

```
=== BUILD TARGET teamreps OF PROJECT teamreps WITH CONFIGURATION Debug ===

Check dependencies
CodeSign error: entitlements are required for product type 'Application' in SDK 'Simulator - iOS 9.2'. Your Xcode installation may be damaged.

** BUILD FAILED **

The following build commands failed:
	Check dependencies
(1 failure)
Command xcodebuild failed with exit code 65
```

I have had this error before, and found that dropping down to Xcode 7.1.1 does in fact let me build again, however I start to receive the `Error: Unable to start service com.apple.debugserver.` error.

**Comment by firescript** [comment]
I installed xcode 7.2, then re-insalled xcode 7.1.1 and it seems to have fixed itself. Not sure what happened, will close for now.

**Comment by firescript** [comment]
It seems eventually I get the error above and the only way i know how to fix it is to upgrade to 7.2 then downgrade back to 7.1.1.. @Mitko-Kerezov any idea?

**Comment by Mitko-Kerezov** [comment]
Hi @firescript,
The error you've received `Your Xcode installation may be damaged` leads me to believe that there may have been some error in the installation or presumably in signing in with your Apple ID.

You said that you are able to execute `tns run ios --release` successfully but are not able to `tns build ios`. Which mainly means that for some reason you are able to produce a build for a device and are not able to build for a simulator. For the sake of this claim you can try executing `tns build ios --for-device` and see if that succeeds.

As to why you can't build for a simulator - I can't really tell, my advice would be a fresh, clean installation of xcode 7.2. After installing make sure you've signed into your Apple ID account. This should suffice for producing a successful build for a simulator.

If it is debugging you're after you could also try debugging on the device instead of an emulator.
